tests/python/unittest/test_image.py

```diff
@@ -92,6 +94,26 @@ def test_imdecode(self):
             cv_image = cv2.imread(img)
             assert_almost_equal(image.asnumpy(), cv_image)
 
+    def test_imdecode_bytearray(self):
```

Comment:
I think the tests should target exception handling. 
Can you please also add a test that assertRaises if the buf is empty?

Reply:
I added two test cases, 2nd one is testing empty buffer case: test_imdecode_empty_buffer